### ISSUE TITLE
Ability to choose location of Story Library folder location.

### ISSUE THREAD

**Issue by CobaltMongoose** [klembot/twinejs#1158]
### Is your feature request related to a problem? Please describe.

Currently the default and unchangeable location of the Story Library (on Windows 10) is under the user Documents folder on the System Boot Drive (C:). I have a smaller SSD main boot drive and I really don't like using it for any kind of project data storage. Text is not so bad but Images and sound files can be a problem.

### Describe the solution you'd like.

Please allow the user the option to choose the location where the Story Library folder is, including allowing the user to specify a different drive other than the system boot drive.

### Describe alternatives you've considered.

The alternative of using (on Windows 10) "mklink" command to create a Symbolic Link to replace the Stories folder with a reference to another folder is generally messy. Symbolic links are not a beginner friendly solution and can cause problems for an unaware user since they appear like a real file folder in Windows Explorer, leading to confusion later in time.

### Additional context on this suggestion.

I am thankful that the installation process did allow me to select the location for the application folder.
Also, the application itself is amazing, may thanks to everyone working on it!

### Presubmission checklist

- [ ] I am interested in working on code that would implement this feature request. (This is not required to submit a suggestion.)
- [X] I have done a search and believe that an issue does not already exist for this idea in the GitHub repository.
- [X] I have read and agree to abide by this project's Code of Conduct.

**Comment by greyelf** [comment]
> but Images and sound files can be a problem.

The general advise is **not** to store any other files (like Images and/or Audio files) need within the folders generated by the Twine 2.x application.

Instead you create your own project related folder structure elsewhere on your drive(s), and then save the HTML file you generate using the application's **Publish to File** option into the parent folder of that structure.

eg, If your project is named **Adventure** and you name the generated HTML file **adventure.html** then a Windows based structure might look something like...
```
D:\adventure\
	audio\
		traffic.mp3
		wind.mp3
	images\
		city.jpg
		forest.jpg
	adventure.html
```

**Comment by greyelf** [comment]
I agree that the ability to choose the location that a Project HTML file is stored in, either for all projects or at an individual project level, would be a great feature to have. The purposed of my previous post was to describe the technique many Authors use to be able to use a Relative URL to access external media files from within a Story HTML they are working on.

note:
The application's **Test**, **Play**, and "**Test story starting here**" options all create a temporary HTML file, and this file will be stored within the default temporary file area of your Operating System (OS).

So external media files you store within the application's project related folders will not be referenceable by a Relative URL. You would need to store those media file within the same OS temporary file area, which is not a good idea because the OS deletes files from that area as needed.

Media files stored within the project related folders are referenceable via a Relative URL if you use the "Show Story Library" option to locate your project's Project HTML file, and then option that HTML file in a web-browser. However I would advise against doing that as the purpose of that HTML file is to store the Passage meta-data of your project, and application expects to have total access to update that file as needed.

**Comment by klembot** [comment]
I'm generally on board with this idea but some of the UX aspects have vexed me.

- What should happen if the user-chosen directory is not accessible when the app starts? e.g. if I set my story library to a removable drive and I don't have it connected, what should Twine do?
  - Twine could fall back to the standard location but that means users have to deal with two story libraries and merging them together at some point.
  - Twine could give up and quit on start but that doesn't feel friendly.
- How does the user set the story library directory? A menu item, probably? Twine will need to restart if the location is changed.

**Comment by hituro** [comment]
I don't think you should merge story libraries, just have one active library, and allow the user to choose the location. If they want to merge libraries, they can move files from one to another. If the current library isn't there on startup, it should ask for a library folder and allow the user to choose.

**Comment by klembot** [comment]
General question, would folks interested in this feature be OK if it was exposed via an environment variable or command-line switch specified at startup time? The reason I ask is this would lower the implementation complexity by a lot, but it means extra work for end users. On Windows, I believe you can [set CLI switches on an app shortcut fairly easily](https://www.addictivetips.com/windows-tips/command-line-switch-on-windows-10/) (and I'm guessing it's just as plausible, if not more, on Linux). However, on macOS it's not as straightforward. I'm wondering if this is an acceptable tradeoff as it feels unscientifically like a feature that power users are asking for.

**Comment by hituro** [comment]
I don't think it's much harder on OSX, you would just do it through the terminal, but that would probably make Twine the *only* non-CLI application I invoked through the terminal. Given that there is no other case in which Twine requires such an approach to run, I've got to say that while I would be able to use such an option, it seems to set a pretty bad precedent.

I don't think this is as much of a power-user feature as you think, since it comes up fairly often, but I will be the first to admit that I don't have a solid piece of proof of that opinion.

**Comment by hawkerm** [comment]
> What should happen if the user-chosen directory is not accessible when the app starts?

The same could be true for the current My Documents location. It's possible to remap the My Documents folder to be auto-synced with One Drive or on a remote network drive instead... (Think there was another issue earlier I saw being linked to around making assumptions about where the My Documents folder is.)

I sync My Documents with One Drive, so I generally keep only certain things there and then have other folders elsewhere for on going work.

Surprised that Twine doesn't already have a settings dialog and other options to change (like default story format for new stories)... oh there is a `Preferences` it's just not in the Menu Bar menu. Can't there be a folder path shown there and a button to change it? (oh, and the default story format is in the other toolbar....)

Still familiarizing myself with Twine, but this is the first thing I saw when I read the intro help boxes and was like I don't want auto-save to save there...
